### PR TITLE
feat(languages): Support markdoc

### PR DESCRIPTION
**Pull request by matoous**
Add basic support for [markdoc](https://markdoc.dev/).

For language server, see: https://github.com/markdoc-extra/markdoc-ls
For tree sitter, see: https://github.com/markdoc-extra/tree-sitter-markdoc

Fixes: https://github.com/helix-editor/helix/issues/6229

### CODE REVIEW

File: runtime/queries/markdoc/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,16 @@
+tag_name: (identifier) @tag
+(tag_self_closing "/" @tag)
+(tag_close "/" @tag)
+([(tag_start) (tag_end) "="] @tag)
+(attribute [key : (identifier)] @attribute)
+(attribute [shorthand : (identifier)]  @attribute)
+(variable [variable : (identifier) (variable_sigil)] @variable)
+(variable_tail property : (identifier) @property)
```
Comment (erasin) [inline runtime/queries/markdoc/highlights.scm:8]: `property` maybe is `attribute` or `variable.other.member`


- https://docs.helix-editor.com/themes.html#syntax-highlighting

**Comment by erasin** [comment]
> For language server, see: https://github.com/markdoc-extra/markdoc-ls

There is the language server installation wiki. we can append this lsp.
https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers

**Comment by matoous** [comment]
@erasin yes yes, I was testing it with the LSP locally, will add it to the wiki once (if) this is merged.

File: languages.toml [hunk 2]
```diff
@@ -2371,6 +2371,7 @@ name = "smithy"
 source = { git = "https://github.com/indoorvivants/tree-sitter-smithy", rev = "cf8c7eb9faf7c7049839585eac19c94af231e6a0" }
 
 [[language]]
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (matoous) [reply languages.toml]: Yes, sorry, done.

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**